enclave-runner/src/usercalls/mod.rs

```diff
@@ -336,59 +343,292 @@ impl EnclaveState {
         })
     }
 
+    fn create_worker_threads(scope : &crossbeam::thread::Scope ,num_of_threads: usize,
+                             work_receiver: crossbeam::crossbeam_channel::Receiver<Work>,
+                             io_queue_send: mpsc::Sender<UsercallSendData>,
+                             enclave: Arc<EnclaveState>) {
+        for _ in 1..num_of_threads {
+            let work_receiver = work_receiver.clone();
+            let io_queue_send = io_queue_send.clone();
+            let enclave = enclave.clone();
+
+            scope.spawn(move |_| {
+                loop {
+                    let work = work_receiver.recv();
+                    let work = match work {
+                        Err(crossbeam::channel::RecvError) => {
+                            break;
+                        },
+                        Ok(work) => work,
+                    };
+                    let res = match work {
+                        Work::Stopped(tcs, mode) => {
+                            RunningTcs::entry_async(enclave.clone(), tcs, &io_queue_send, mode)
+                        },
+                        Work::Running(state, usercall, coresult, mode) => {
+                            RunningTcs::coentry_async(state, usercall, coresult, &io_queue_send, mode)
+                        },
+                    };
+                    if let Err(err) = res {
```

Comment:
This code seems... weird. Why are you putting threads back on the library queue after they return an error? And AFAICT threads are never put back on any queue when succesful.